### PR TITLE
feat(crux): use proxy headers to determine the node ip

### PR DESCRIPTION
**Pull request by m8vago**
Take proxy headers into account while determining the address of a node.

### CODE REVIEW

File: web/crux/src/main.ts [hunk 1]
```diff
@@ -58,7 +56,7 @@ const bootstrap = async () => {
   const document = SwaggerModule.createDocument(app, config)
   SwaggerModule.setup('/api/swagger', app, document)
 
-  const agentOptions = loadGrpcOptions('agent', configService.get<string>('GRPC_AGENT_PORT'))
```
Comment (polaroi8d) [inline web/crux/src/main.ts:61]: 🦅

File: web/crux/src/shared/grpc-node-connection.ts [hunk 2]
```diff
@@ -97,6 +101,15 @@ export default class GrpcNodeConnection {
     this.statusChannel.complete()
     this.call.connection = null
   }
+
+  private getFirstItemOfStringArrayMetadata(key: string): string | null {
```
Comment (polaroi8d) [inline web/crux/src/shared/grpc-node-connection.ts:105]: Tests? 🤔

File: web/crux/src/main.ts (hunk 1 above)
Comment (m8vago) [reply web/crux/src/main.ts:61]: :man_shrugging:

File: web/crux/src/shared/grpc-node-connection.ts (hunk 2 above)
Comment (m8vago) [reply web/crux/src/shared/grpc-node-connection.ts:105]: it's a private method

Comment (polaroi8d) [reply web/crux/src/shared/grpc-node-connection.ts:105]: https://stackoverflow.com/a/56199483/5804254 ?

**Review by polaroi8d — APPROVED** [review]
LGTM

**Pull request merged**